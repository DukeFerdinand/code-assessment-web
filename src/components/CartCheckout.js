import React from 'react'
import PropTypes from 'prop-types'

import '../styles/components/CartCheckout.scss'

export default class CartCheckout extends React.Component {
  constructor(props) {
    super(props)

    this.totals = this.findTax(props.total)
  }
  findTax(total) {
    return {
      subtotal: total,
      tax: (total * .08).toFixed(2),
      get total() {
        return (parseFloat(this.subtotal) + parseFloat(this.tax)).toFixed(2)
      }
    }
  }
  render() {
    const { subtotal, tax, total } = this.totals
    return (
      <div className="checkout-wrapper">
        <hr className="divider" />
        <div className="checkout-row">
          <p className="checkout-row-title">Subtotal</p>
          <p className="checkout-row-price checkout-subtotal">&#36;{subtotal}</p>
        </div>
        <div className="checkout-row">
          <p className="checkout-row-title">Tax</p>
          <p className="checkout-row-price checkout-tax">&#36;{tax}</p>
        </div>
        <hr className="divider" />
        <div className="checkout-row">
          <p className="checkout-row-title">Total</p>
          <p className="checkout-row-price checkout-total">&#36;{total}</p>
        </div>
        <div className="checkout-row checkout-row-final">
          <button className="checkout-button" onClick={this.props.onCheckoutClicked}>
            CHECKOUT
          </button>
        </div>
      </div>
    )

  }
}

CartCheckout.PropTypes = {
  total: PropTypes.number.isRequired,
  onCheckoutClicked: PropTypes.func.isRequired
}