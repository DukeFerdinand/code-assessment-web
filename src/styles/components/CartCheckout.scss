@import '../shared';

.checkout-wrapper {
  // background: red;
  // display: flex;
  // flex-direction: column;
  .checkout-row {
    flex: 1;
    display: flex;
    justify-content: space-between;
    margin: 25px 0;

    // @media (min-width: $desktop) {
    //   background: red;
    // }

    .checkout-row-title {
      color: $text-color-secondary - 20;
    }
    .checout-row-price {

    }

    &.checkout-row-final {
      margin: 0;
    }

  }
  .checkout-button {
    // position: relative;
    // left: 0;
    // right: 0;
    font-size: .90em;
    background: $button-primary;
    cursor: pointer;
    color: white;
    width: 130px;
    padding: 10px;
    margin: 0 auto;
    border-radius: 25px;

    &:hover, &:focus {
      background: $button-primary-hover
    }
  }
}