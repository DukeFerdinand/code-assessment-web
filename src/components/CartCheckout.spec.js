import React from 'react'
import { shallow } from 'enzyme'
import CartCheckout from './CartCheckout'

const setup = total => {
  const actions = {
    onCheckoutClicked: jest.fn()
  }

  const component = shallow(<CartCheckout total={total} {...actions} />)

  return {
    component,
    subtotal: component.find('p.checkout-subtotal'),
    tax: component.find('p.checkout-tax'),
    total: component.find('p.checkout-total')
  }
}

describe('Cart Checkout component', () => {
  describe('when given total', () => {
    it('should render subtotal, tax, and total', () => {
      const { subtotal, tax, total } = setup(500.01)
      expect(subtotal.text()).toBe('$500.01');
      expect(tax.text()).toBe('$40.00') // component assumes .08 tax rate
      expect(total.text()).toBe('$540.01')
    })
  });
  it('should match snapshot', () => {
    const { component } = setup('$500.01')
    expect(component).toMatchSnapshot()
  })
});