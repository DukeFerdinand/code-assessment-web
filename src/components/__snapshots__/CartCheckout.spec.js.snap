// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Checkout component should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="checkout-wrapper"
  >
    <hr
      className="divider"
    />
    <div
      className="checkout-row"
    >
      <p
        className="checkout-row-title"
      >
        Subtotal
      </p>
      <p
        className="checkout-row-price checkout-subtotal"
      >
        $
        $500.01
      </p>
    </div>
    <div
      className="checkout-row"
    >
      <p
        className="checkout-row-title"
      >
        Tax
      </p>
      <p
        className="checkout-row-price checkout-tax"
      >
        $
        NaN
      </p>
    </div>
    <hr
      className="divider"
    />
    <div
      className="checkout-row"
    >
      <p
        className="checkout-row-title"
      >
        Total
      </p>
      <p
        className="checkout-row-price checkout-total"
      >
        $
        NaN
      </p>
    </div>
    <div
      className="checkout-row checkout-row-final"
    >
      <button
        className="checkout-button"
        onClick={[MockFunction]}
      >
        CHECKOUT
      </button>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="checkout-wrapper"
    >
      <hr
        className="divider"
      />
      <div
        className="checkout-row"
      >
        <p
          className="checkout-row-title"
        >
          Subtotal
        </p>
        <p
          className="checkout-row-price checkout-subtotal"
        >
          $
          $500.01
        </p>
      </div>
      <div
        className="checkout-row"
      >
        <p
          className="checkout-row-title"
        >
          Tax
        </p>
        <p
          className="checkout-row-price checkout-tax"
        >
          $
          NaN
        </p>
      </div>
      <hr
        className="divider"
      />
      <div
        className="checkout-row"
      >
        <p
          className="checkout-row-title"
        >
          Total
        </p>
        <p
          className="checkout-row-price checkout-total"
        >
          $
          NaN
        </p>
      </div>
      <div
        className="checkout-row checkout-row-final"
      >
        <button
          className="checkout-button"
          onClick={[MockFunction]}
        >
          CHECKOUT
        </button>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CartCheckout
        onCheckoutClicked={[MockFunction]}
        total="$500.01"
      />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CartCheckout {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onCheckoutClicked": [MockFunction],
          "total": "$500.01",
        },
        "refs": Object {},
        "state": null,
        "totals": Object {
          "subtotal": "$500.01",
          "tax": "NaN",
          "total": "NaN",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="checkout-wrapper"
        >
          <hr
            className="divider"
          />
          <div
            className="checkout-row"
          >
            <p
              className="checkout-row-title"
            >
              Subtotal
            </p>
            <p
              className="checkout-row-price checkout-subtotal"
            >
              $
              $500.01
            </p>
          </div>
          <div
            className="checkout-row"
          >
            <p
              className="checkout-row-title"
            >
              Tax
            </p>
            <p
              className="checkout-row-price checkout-tax"
            >
              $
              NaN
            </p>
          </div>
          <hr
            className="divider"
          />
          <div
            className="checkout-row"
          >
            <p
              className="checkout-row-title"
            >
              Total
            </p>
            <p
              className="checkout-row-price checkout-total"
            >
              $
              NaN
            </p>
          </div>
          <div
            className="checkout-row checkout-row-final"
          >
            <button
              className="checkout-button"
              onClick={[MockFunction]}
            >
              CHECKOUT
            </button>
          </div>
        </div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="checkout-wrapper"
        >
          <hr
            className="divider"
          />
          <div
            className="checkout-row"
          >
            <p
              className="checkout-row-title"
            >
              Subtotal
            </p>
            <p
              className="checkout-row-price checkout-subtotal"
            >
              $
              $500.01
            </p>
          </div>
          <div
            className="checkout-row"
          >
            <p
              className="checkout-row-title"
            >
              Tax
            </p>
            <p
              className="checkout-row-price checkout-tax"
            >
              $
              NaN
            </p>
          </div>
          <hr
            className="divider"
          />
          <div
            className="checkout-row"
          >
            <p
              className="checkout-row-title"
            >
              Total
            </p>
            <p
              className="checkout-row-price checkout-total"
            >
              $
              NaN
            </p>
          </div>
          <div
            className="checkout-row checkout-row-final"
          >
            <button
              className="checkout-button"
              onClick={[MockFunction]}
            >
              CHECKOUT
            </button>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CartCheckout
    onCheckoutClicked={[MockFunction]}
    total="$500.01"
  />,
}
`;
